src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -377,6 +377,14 @@ suggestExportUnusedTopBinding srcOpt ParsedModule{pm_parsed_source = L _ HsModul
         _ -> False
     needsComma _ _ = False
 
+    opLetter :: String
+    opLetter = ":!#$%&*+./<=>?@\\^|-~"
+
+    parenthesizeIfNeeds :: Bool -> T.Text -> T.Text
+    parenthesizeIfNeeds needsTypeKeyword x 
+      | any (`elem` opLetter) . T.unpack $ x = (if needsTypeKeyword then "type " else "") <> "(" <> x <>")"
```

Comment:
```suggestion
      | any (`elem` opLetter) . take 1 . T.unpack $ x = (if needsTypeKeyword then "type " else "") <> "(" <> x <>")"
```

Reply:
Yes, looking at the first char would be enough.